transcom/mymove
# MB-4061 Filter payment requests by GBLOC

## Description

When a TIO signs in, they should only see the payment
requests associated with a transportation office that has the same
GBLOC as the transporation office the office user belongs to.

This PR also adds authorization to the list payment requests endpoint
so that it only allows users who have the TIO role.

In order to speed up tests and avoid using the DB when not needed, I
added a new top-level `Stub` assertion type to `testdatagen` so that
you can bypass the DB when setting it to `true`. The testdatagen
functions are very convenient, but sometimes we just want plain models
without saving them to the DB. By setting the `Stub` assertion, we can
take advantage of that convenience without having to write the models
from scratch in tests.

## Reviewer Notes

Steps to test:
1. Run the setup commands in the section below
2. Visit http://officelocal:3000/devlocal-auth/login
2. Click the `Login` button next to `tio_role@office.mil (PPM office)`
3. Verify that you can see payment requests
5. Sign in to http://milmovelocal:3000 with your `@truss.works` email that you use for testing
6. Sign in to the admin interface and create a new office user for that same email, and assign it to a transportation office whose GBLOC is not "LKNQ", then add the TOO and TIO roles.
7. Sign in to http://officelocal:3000
8. Click the "change user roles" link in the top left
9. Click the TIO button

Expected Result: you should not see any payment requests

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run (in a separate terminal tab)
```

## Code Review Verification Steps

 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-4061) for this change

## Screenshots

If this PR makes visible UI changes, an image of the finished UI can help reviewers and casual
observers understand the context of the changes. A before image is optional and
can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow instead of using multiple images. You may want to use GIPHY CAPTURE for this! 📸

_Please frame screenshots to show enough useful context but also highlight the affected regions._
